Sysbench Io Benchmarking

tsebastiani opened the pull request:
parsing function and schema update for Random output


added unit tests


unit tests

## Changes introduced with this PR

Enabling the sysbench io benchmark functionality for chaos purpose.

---
By contributing to this repository, I agree to the [contribution guidelines](https://github.com/arcalot/.github/blob/main/CONTRIBUTING.md).

dustinblack commented on arcaflow_plugin_sysbench/sysbench_schema.py:
```diff
@@ -223,6 +249,116 @@ class SysbenchMemoryInputParams(CommonInputParameters):
+class SysbenchIoInputParams(CommonInputParameters):
+    """
+    This is the data structure for the
+    input parameters of Sysbench CPU benchmark.
```
Extremely minor nit -- Should be "IO benchmark" not "CPU benchmark"

tsebastiani replied:
fixed and pushed.

dustinblack commented on arcaflow_plugin_sysbench/sysbench_schema.py:
```diff
@@ -433,6 +626,37 @@ class SysbenchCpuResultParams:
+    Latency: typing.Annotated[
+        LatencyAggregates,
+        schema.name("Latency"),
+        schema.description("CPU latency in miliseconds"),
```
Is this actually CPU latency? Or file operation latency?

tsebastiani replied:
I've no clue about that, what I know is that the output structure is the same as for the other tests, so I supposed it was the same kind of "Latency".

dustinblack replied:
It's hard to find an absolutely clear answer, and my C skills are pretty atrophied, but I feel very sure that this latency value is for the file operation, not the CPU.

dustinblack reviewed (changes requested):
Very minor nits

Harshith-umesh reviewed (commented):
@tsebastiani could you please add an example config for the io tests here: https://github.com/arcalot/arcaflow-plugin-sysbench/tree/io_benchmark/configs

tsebastiani committed:
io benchmark, schemas  logic and unit tests in place + pyproject.toml fix


parsing function and schema update for Random output


added unit tests


unit tests


fixed comment length


fixed optional values in output issue


fixed comment

tsebastiani committed:
added configuration example

tsebastiani commented:
> @tsebastiani could you please add an example config for the io tests here: https://github.com/arcalot/arcaflow-plugin-sysbench/tree/io_benchmark/configs

added basic configuration example

Harshith-umesh reviewed (approved):
LGTM!

tsebastiani committed:
removed CPU reference from IO latency


milliseconds
